### PR TITLE
Fixing CI by xfail on new svd related failure

### PR DESCRIPTION
**Pull request by vfdev-5**
Description:
- Temporary fix CI with `xfail` on new svd related failure
- replace `_svd_helper` by `_linalg_svd` to reenable vmap
- removed rsub_rsub_scalar to fix unexpected failure

related https://github.com/pytorch/functorch/issues/428

### CODE REVIEW

File: test/test_ops.py [hunk 1]
```diff
@@ -369,7 +369,13 @@ def wrapped_fn(*args, **kwargs):
...
         xfail('linalg.eigh'),
 
-
+        # Issue with newer svd implementation and forwardAD
+        # https://github.com/pytorch/pytorch/pull/69827
+        # https://github.com/pytorch/pytorch/pull/70253
+        # RuntimeError: tensor.mH is only supported on matrices or batches of matrices. Got 1-D tensor.
+        xfail('linalg.cond'),
+        xfail('linalg.svdvals'),
+        xfail('norm', 'nuc'),
```
Comment (zou3519) [inline test/test_ops.py:378]: (just taking notes) this is not a regression

File: test/test_ops.py [hunk 2]
```diff
@@ -575,6 +581,13 @@ def vjp_of_vjp(*args_and_cotangents):
         xfail('index_put'),
         xfail('lu_solve'),
         xfail('nn.functional.instance_norm'),
+
+        # Issue with newer svd implementation and forwardAD
+        # https://github.com/pytorch/pytorch/pull/69827
+        # https://github.com/pytorch/pytorch/pull/70253
+        # RuntimeError: vmap: inplace arithmetic(self, *extra_args) is not possible because there exists
+        xfail('linalg.svd'),
+        xfail('svd'),
```
Comment (zou3519) [inline test/test_ops.py:590]: (just taking notes) this is a regression

File: test/test_ops.py [hunk 3]
```diff
@@ -759,6 +780,14 @@ def test_vmapjvp(self, device, dtype, op):
...
         # Runtime Error: The tangent part of the matrix A should also be symmetric.
         xfail('linalg.eigh'),
+
+        # Issue with newer svd implementation and forwardAD
+        # https://github.com/pytorch/pytorch/pull/69827
+        # https://github.com/pytorch/pytorch/pull/70253
+        # RuntimeError: tensor.mH is only supported on matrices or batches of matrices. Got 1-D tensor.
+        xfail('linalg.cond'),
+        xfail('linalg.svdvals'),
+        xfail('norm', 'nuc'),
```
Comment (zou3519) [inline test/test_ops.py:790]: (just taking notes) this is not a regression

File: test/test_ops.py [hunk 4]
```diff
@@ -871,6 +900,15 @@ def test_vmapjvpall(self, device, dtype, op):
...
         xfail('linalg.tensorsolve'),
+
+        # Issue with newer svd implementation and forwardAD
+        # https://github.com/pytorch/pytorch/pull/69827
+        # https://github.com/pytorch/pytorch/pull/70253
+        # RuntimeError: vmap: inplace arithmetic(self, *extra_args) is not possible because there
+        xfail('linalg.svd'),
+        xfail('svd'),
+        xfail('linalg.svdvals'),
+        xfail('norm', 'nuc'),
```
Comment (zou3519) [inline test/test_ops.py:911]: (just taking notes) this is a regression

File: test/test_ops.py [hunk 5]
```diff
@@ -920,6 +957,14 @@ def test():
...
         xfail('nn.functional.fractional_max_pool2d'),
         skip('solve'),
+
+        # Issue with newer svd implementation and forwardAD
+        # https://github.com/pytorch/pytorch/pull/69827
+        # https://github.com/pytorch/pytorch/pull/70253
+        #
+        # RuntimeError: tensor.mH is only supported on matrices or batches of matrices. Got 1-D tensor.
+        xfail('linalg.cond'),
+        xfail('linalg.svdvals'),
```
Comment (zou3519) [inline test/test_ops.py:967]: (just taking notes) this is a regression

File: test/test_ops.py (hunk 2 above)
Comment (zou3519) [reply test/test_ops.py:590]: This can be fixed by modifying the in-place operations in svd_backward

File: test/test_ops.py (hunk 5 above)
Comment (Lezcano) [reply test/test_ops.py:967]: This one was working before but throwing a warning saying that "we're defaulting to an iterative method note that this is inefficient" or smth like that. Note that `linalg.svdvals` did not and does not have an unpacking rule before, and we have not changed its implementation anyway, so it's not clear to me why it was falling back to an iterative method that made it work before and now it is not. Any ideas?

Comment (vfdev-5) [reply test/test_ops.py:967]: `xfail('linalg.svdvals')` was moved from line 910 to 967, maybe not a regression as once we started decomposing `linalg_svdvals` it was directly xfailed.

**Comment by zou3519** [comment]
@vfdev-5 could you rebase this when you get a chance please?

**Comment by vfdev-5** [comment]
@zou3519 can you detail what you would like to do ? `main` branch is already in the PR, done by @Chillee , PR is up-to-date vs `main`.

However, I was expecting that if @Leszano fixed pytorch side and it was merged we can close this PR

**Comment by zou3519** [comment]
@vfdev-5 thanks for the explanation, I didn't realize Horace rebased this already!

**Comment by zou3519** [comment]
From what I can tell:
- pytorch nightlies haven't been updated with the fix yet
- When it does update with the fix, we only really need to merge one line from this change (the line that changes _svd_helper to _linalg_svd). So we can just open a new PR for that or we can include it as a part of another fix

So TL;DR no action required from us yet

**Comment by vfdev-5** [comment]
Yes, this line `_svd_helper -> _linalg_svd` and probably `norm_nuc` was not fixed

**Comment by Lezcano** [comment]
Also look for the appearances of `linalg.svd`  in the xfails and skips and remove those, as those test pass now (there's 1xfail and 1 skip, at least)

**Comment by zou3519** [comment]
I incorporated the change and xfails while fixing the rest of the functorch CI in https://github.com/pytorch/functorch/pull/452

**Comment by vfdev-5** [comment]
Thanks! OK, let's close this PR in favor of #452

**Pull request closed without merging**